File: worker/webrtc/rtc.go
package webrtc

type InputPacket struct {
	Type string `json:"type"`
	Data string `json:"data"`
}

type MouseEventPayload struct {
	IsLeft byte    `json:"isleft"`
	X      float32 `json:"x"`
	Y      float32 `json:"y"`
	Width  float32 `json:"w"`
	Height float32 `json:"h"`
}

type KeyEventPayload struct {
	KeyCode byte `json:"keycode"`
}

File: web/src/api/rtc.ts
export type InputPacket = {
  type : string;
  data: string;
}

export type MouseEventPayload = {
  isleft: number;
  x: number;
  y: number;
  w: number;
  h: number
}

export type KeyEventPayload = {
  keycode: number;
}
export enum EventType {
  KEY_UP     = "KEYUP",
  KEY_DOWN   = "KEYDOWN",
  MOUSE_UP   = "MOUSEUP",
  MOUSE_DOWN = "MOUSEDOWN",
  MOUSE_MOVE = "MOUSEMOVE"
}